src/rgw/rgw_dedup.cc

```diff
@@ -757,6 +919,16 @@ namespace rgw::dedup {
                        << p_tgt_rec->s.md5_low << std::dec << dendl;
   }
 
+  //---------------------------------------------------------------------------
```

Comment:
As I assume performance is an issue here:
the following version of stringToHex() benches at about 10X faster
when tested in my environment:

std::string stringToHex(const std::string& input)
{
    fmt::memory_buffer buf;
    buf.reserve(input.size() * 2);
    for (const unsigned char b : input) {
        fmt::format_to(std::back_inserter(buf), FMT_COMPILE("{:02X}"), b);
    }
    return string{buf.begin(), buf.end()};
}

Reply:
Thanks, I replaced the function with your version